securesystemslib/metadata.py

```diff
@@ -161,3 +171,28 @@ def verify(self, keys: List[Key], threshold: int) -> Dict[str, Key]:
             )
 
         return accepted_keys
+
+    def deserialize_payload(
+        self,
+        class_type: Any,
+        deserializer: Optional[BaseDeserializer] = None,
+    ) -> Any:
+        """Parse DSSE payload.
+
+        Arguments:
+            class_type: A class having a from_dict method.
```

Comment:
This only has to be `from_dict` if we use default `JSONDeserializer`. For other deserializers `class_type` may not even be needed. Should we make it optional?

Reply:
`BaseDeserializer` requires a class type to store the bytes into a containerized form. That's why the class type will be required by every deserializer.

Reply:
My comment about `from_dict` ist still true. Not every `deserializer` might require the class to have a `from_dict` method. What about something like...
```suggestion
            class_type: The class to  be deserialized. If the default deserializer is
                used, it must implement ``JSONSerializable``.
```